yaml/yaml_test.go

```diff
@@ -113,3 +113,119 @@ func TestArray(t *testing.T) {
 	assert.Equal(t, verify[0]["c"], 3)
 	assert.Equal(t, verify[1]["c"], 4)
 }
+
+func TestEmptyArrayIntoInterface(t *testing.T) {
+	input := []byte(`
+a: []
+`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	verify := struct {
+		A interface{}
+	}{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Nil(t, verify.A)
+}
+
+func TestEmptyArrayInStructIntoArrayOfInterfaces(t *testing.T) {
+	input := []byte(`
+a: []
+`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	verify := struct {
+		A []interface{}
+	}{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Equal(t, []interface{}{}, verify.A)
+}
+
+func TestEmptyArrayIntoArrayOfInterfaces(t *testing.T) {
+	input := []byte(`[]`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	var verify []interface{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Equal(t, []interface{}{}, verify)
+}
+
+func TestEmptyMapIntoInterface(t *testing.T) {
+	input := []byte(`{}`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	verify := struct {
+		A interface{}
+	}{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Nil(t, verify.A)
+}
+
+func TestEmptyMapInStructIntoInterface(t *testing.T) {
+	input := []byte(`
+a: {}
+`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	verify := struct {
+		A interface{}
+	}{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Nil(t, verify.A)
+}
+
+func TestEmptyMapInStructIntoMapOfInterfaces(t *testing.T) {
+	input := []byte(`
+a: {}
+`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	verify := struct {
+		A map[string]interface{}
+	}{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Equal(t, map[string]interface{}{}, verify.A)
+}
+
+func TestEmptyMapIntoMapOfInterfaces(t *testing.T) {
+	input := []byte(`{}`)
+
+	c, err := NewConfig(input)
+	if err != nil {
+		t.Fatalf("failed to parse input: %v", err)
+	}
+
+	var verify map[string]interface{}
+	err = c.Unpack(&verify)
+	assert.Nil(t, err)
+	assert.Equal(t, map[string]interface{}{}, verify)
+}
```

Comment:
there is quite some repetition here. How about moving the test cases into a table like so?

```
cases := map[string]struct{
  input string
  to interface{}
  want interface{}
}{
  "empty map into map of interfaces": {
    input: "{}",
    to: &map[string]interface{}{},
    want: &map[string]interface{}{},
  },
  ...
}

for name, test := range cases {
  t.Run(name, func(t *testing.T) {
    c, err := MustNewConfig(byte(test.input))
    mustUnpack(t, c, test.to)
    assert.Equal(t, test.want, test.to)
  })
}
```

Reply:
Fixed.

Reply:
`mustUnpack` should be available already if you rebase.

Reply:
Rebased and refactored as it wasn't exposed.